### DISCUSSION TITLE
[QUESTION] Advantages compare to existing alternates such as lsd and eza

### DISCUSSION THREAD

**Discussion by CarterLi** [Equationzhao/g#172] [Q&A] [answered]
I'm an eza user and new to this project. It would be awesome to see some feature and performance comparation between eza and g-ls. Thanks.

**Comment by equationzhao** [comment, marked as answer]
😂 since you've been using eza, I'll recommend you sticking with it. 

`eza` can provide better performance than `g` up to 3x on macOS.
(as I tested, `eza` will take 5ms/120ms for 20/10000 files, and `g` will take 10ms/400ms, FYI)

well, `g` does have some unique features like 
1. listing files' mime-type/charset
2. listing files' modified, created, accessed time together (while `eza` seems to only list one of those fields at a time)
3. listing files in table/markdown/json/csv/tsv view
4. show files' relative path to a specific directory
5. customizable theme (not sure if `eza` has similar settings 🤔, but `g` does provide the ability to customize your theme)

I started the project as a way to practice my Go skills, and definitely it has some weakness like the lack of completion(only zsh is supported yet), lack of test. But I'll try to improve the project~


btw, I've been using fastfetch for a long time, thanks for your awesome project 🥳

**Reply by flawrist** [reply]
g v0.3.0 takes up 16.7 MiB, it is insane even given that it's statically compiled with `-s -w` flags.

Comparison with other Go-based apps:
``` console
$ g -1 --size
 3.8 MiB age.exe
 9.4 MiB dnscrypt-proxy.exe
11.8 MiB doggo.exe
 6.7 MiB f2.exe
 3.0 MiB goawk.exe
```

**Reply by equationzhao** [reply]
Yeah, I've noticed that and have tried several ways to reduce the binary size, but so far haven't found a really promising solution.

Currently, I'm using the upx tool to compress the binaries for the release files, which only works on Linux and Windows. After compression, the size drops to about 5-6 MiB on those platforms.

This is definitely something that needs to be addressed. If you're building from source yourself, you can also try using upx to compress the binary before this issue #237 is properly resolved.

**Comment by CarterLi** [comment]
Ok, thanks for your reply.
